src/openrct2/scripting/bindings/world/ScTileElement.cpp

```diff
@@ -2040,13 +2079,63 @@ namespace OpenRCT2::Scripting
         duk_push_uint(ctx, _element->GetOwner());
         return DukValue::take_from_stack(ctx);
     }
-
     void ScTileElement::owner_set(uint8_t value)
     {
         ThrowIfGameStateNotMutable();
         _element->SetOwner(value);
     }
 
+    DukValue ScTileElement::text_get() const
+    {
+        auto& scriptEngine = GetContext()->GetScriptEngine();
+        auto* ctx = scriptEngine.GetContext();
+        BannerIndex idx = _element->GetBannerIndex();
+        if (idx == BannerIndex::GetNull())
+            duk_push_null(ctx);
+        else
+            duk_push_string(ctx, GetBanner(idx)->GetText().c_str());
+        return DukValue::take_from_stack(ctx);
+    }
+    void ScTileElement::text_set(std::string value)
+    {
+        ThrowIfGameStateNotMutable();
+        BannerIndex idx = _element->GetBannerIndex();
+        if (idx != BannerIndex::GetNull())
+        {
+            GetBanner(idx)->text = value;
+            ScrollingTextInvalidate();
```

Comment:
You shouldn't have to call this, the cache is currently busted.

Reply:
Actually the text does not properly update for LS and walls, no idea why.

Reply:
I removed it and the text also updates now properly.